dts/bindings/memory-controllers/atmel,sam-smc.yaml

```diff
@@ -0,0 +1,167 @@
+# Copyright (c) 2022, Basalte bv
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  Atmel Static Memory Controller (SMC).
+
+  The SMC allows to interface with static-memory mapped external devices such as
+  SRAM, NOR Flash, NAND Flash, SDRAM...
+
+  The SMC controller can have up to 4 children defining the connected external
+  memory devices. Device Tree example:
+
+  &smc {
+    status = "okay";
+    pinctrl-0 = <&pc18a_smc_a0
+          &pc19a_smc_a1
+          &pc20a_smc_a2
+          &pc21a_smc_a3
+          &pc22a_smc_a4
+          &pc23a_smc_a5
+          &pc24a_smc_a6
+          &pc25a_smc_a7
+          &pc26a_smc_a8
+          &pc27a_smc_a9
+          &pc28a_smc_a10
+          &pc29a_smc_a11
+          &pc30a_smc_a12
+          &pc31a_smc_a13
+          &pa18c_smc_a14
+          &pa19c_smc_a15
+          &pa20c_smc_a16
+          &pa0c_smc_a17
+          &pa1c_smc_a18
+          &pc0a_smc_d0
+          &pc1a_smc_d1
+          &pc2a_smc_d2
+          &pc3a_smc_d3
+          &pc4a_smc_d4
+          &pc5a_smc_d5
+          &pc6a_smc_d6
+          &pc7a_smc_d7
+          &pc14a_smc_ncs0
+          &pc15a_smc_ncs1
+          &pc11a_smc_nrd
+          &pc8a_smc_nwe>;
+
+    is66wv51216dbll@0 {
+      reg = <0>;
+
+      atmel,smc-write-mode = "nwe";
+      atmel,smc-read-mode = "nrd";
+      atmel,smc-setup-timing = <1 1 1 1>;
+      atmel,smc-pulse-timing = <6 6 6 6>;
+      atmel,smc-cycle-timing = <7 7>;
+    };
+  };
+
+  Finally, in order to make the memory available you will need to define new
+  memory device/s in DeviceTree:
+
+  sdram1: sdram@60000000 {
+      compatible = "mmio-sram";
+      device_type = "memory";
+      reg = <0x60000000 DT_SIZE_M(X)>;
+      zephyr,memory-region = "SDRAM1";
```

Comment:
The EBI allows you to have SDRAM but uses a dedicated controller which is not SMC. Please, fix the example.